### PR TITLE
Update to 0.15.0 PureScript

### LINKED ISSUE #273: Update to 0.15.0 compiler
Now that `0.15.0` is released, we can update this project. Not sure if we want to wait a bit until the package set gets filled with more packages first, but I imagine that will happen quickly over the next week.

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Fixes #273. Still a WIP. Besides updating the server code, I also updated the client and staging code to `0.15.0`.

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0 by @)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: client/packages.dhall [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
-          , "nullable"
-          , "prelude"
-          , "refs"
-          , "web-html"
-          , "web-uievents"
-          ]
+          [ "console", "effect", "functions", "prelude" ]
         }
-    }
+      }
```
Comment (JordanMartinez) [inline client/packages.dhall:12]: This should be removed once `debug` gets added to the package set.

File: client/spago.dhall [hunk 2]
```diff
@@ -38,8 +46,9 @@
   , "transformers"
   , "tuples"
   , "unfoldable"
+  , "unsafe-coerce"
   , "web-html"
   ]
 , packages = ./packages.dhall
-, sources = [ "src/**/*.purs", "test/**/*.purs" ]
+, sources = [ "src/**/*.purs", "config/dev/**/*.purs", "test/**/*.purs" ]
```
Comment (JordanMartinez) [inline client/spago.dhall:53]: Perhaps we should use 3 `spago.dhall` files: a shared one, one for dev and one for prod?

**Comment by mikesol** [comment]
This looks really good! I tried to hack at it a bit and spotted that `setupIFrame` and `teardownIFrame` are exported from Container.js without being defined there, which may be problematic for `esbuild`. Is that the bundler you're going to use for the `client`?

**Comment by mikesol** [comment]
@JordanMartinez if you want to assign me as a reviewer I can mark the places where I needed to make changes for my dockerized version of trypurescript (the one on try.wags.fm). Some of them will be needed for this PR as well (for example, I hit the `esbuild` issues with `top` being a reserved word, etc).

**Comment by JordanMartinez** [comment]
@mikesol This PR will need to update the code to do both #263 and #264.

> `setupIFrame` and `teardownIFrame` are exported from Container.js without being defined there, which may be problematic for esbuild

Looks like those two functions are defined in `client/public/index.html`. Since we'll be using SystemJS, I don't think we'll implement the code in the same way as before, so this might be a moot point.

**Comment by JordanMartinez** [comment]
Succeeded by #275

**Pull request closed without merging**